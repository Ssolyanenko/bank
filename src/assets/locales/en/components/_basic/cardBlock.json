{
  "cardBlock": {
    "orderACard": "Order a card"
  }
}
