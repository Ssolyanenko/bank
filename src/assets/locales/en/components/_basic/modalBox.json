{
  "modalBox": {
    "codeExpirationTime": "Code expiration time",
    "seconds": "seconds",
    "didNotGetTheCode": "Didnâ€™t get the code?",
    "sendCodeAgain": "Send code again in"
  }
}
