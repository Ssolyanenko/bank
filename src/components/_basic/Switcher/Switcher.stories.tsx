import { useEffect, useState } from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';

import { Switcher } from './Switcher';

export default {
  title: 'Basic/Switcher',
  component: Switcher,
  parameters: {
    docs: {
      description: {
        component: 'Switcher for Forms',
      },
    },
  },
} as ComponentMeta<typeof Switcher>;

const Template: ComponentStory<typeof Switcher> = ({ isChecked }) => {
  const [isCheckedState, setIsCheckedState] = useState(isChecked);

  useEffect((): void => {
    setIsCheckedState(isChecked);
  }, [isChecked]);

  return <Switcher isChecked={isCheckedState} handleChange={(): void => setIsCheckedState(!isCheckedState)} />;
};

export const Default = Template.bind({});

Default.args = {
  isChecked: false,
};
