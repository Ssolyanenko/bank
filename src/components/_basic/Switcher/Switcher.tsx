import React, { FC } from 'react';
import { alpha, styled, Switch } from '@mui/material';

import classes from './Switcher.module.scss';

interface Props {
  handleChange?(): void;
  isChecked?: boolean;
}

const OrangeSwitch = styled(Switch)(({ theme }) => ({
  '& .MuiSwitch-switchBase.Mui-checked': {
    color: '#ffd600',
    '&:hover': {
      backgroundColor: alpha('#ffd600', theme.palette.action.hoverOpacity),
    },
  },
  '& .MuiSwitch-switchBase.Mui-checked + .MuiSwitch-track': {
    backgroundColor: '#ffd600',
  },
}));

export const Switcher: FC<Props> = ({ handleChange, isChecked }) => (
  <div className={classes.switcher}>
    <OrangeSwitch inputProps={{ 'aria-label': 'primary checkbox' }} onChange={handleChange} checked={isChecked} />
  </div>
);
