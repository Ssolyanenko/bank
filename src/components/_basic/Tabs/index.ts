export * from './Tabs';
export * from './TabTitle';
