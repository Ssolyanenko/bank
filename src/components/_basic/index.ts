export * from './PrimaryButton';
export * from './Button';
export * from './Input';
export * from './InputPassword';
export * from './ModalWrapper';
export * from './ModalConfirmation';
export * from './PhoneCodeTimer';
export * from './ModalBox';
export * from './CloseButton';
export * from './BackButton';
export * from './StepperForm';
export * from './StepperLabel';
export * from './ChatButton';
export * from './BanTimer';
export * from './CardTemplate';
export * from './SecondaryButton';
export * from './generic';
export * from './SelectStyled';
export * from './TextFieldStyled';
export * from './InputStyled';
export * from './CardTemplate';
export * from './Modal';
export * from './SecureWrapper';
export * from './InputRange';
export * from './DropDown';
export * from './RadioButtonsGroup';
export * from './FormError';
export * from './Checkbox';
export * from './Switcher';
export * from './EditButton';
export * from './CardBlock';
export * from './Tabs';
export * from './ModalSuccess';
export * from './DropDownWithCheckbox';
export * from './SelectFormikTemplate';
export * from './Loader';
export * from './DatePicker';
export * from './CustomRoutes';
