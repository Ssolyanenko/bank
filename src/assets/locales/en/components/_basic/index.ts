import modalBox from './modalBox.json';
import cardBlock from './cardBlock.json';
import stepperForm from './stepperForm.json';

export const translationsBasic = {
  ...modalBox,
  ...cardBlock,
  ...stepperForm,
};
