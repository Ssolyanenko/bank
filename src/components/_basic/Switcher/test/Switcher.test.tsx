import React from 'react';
import { shallow } from 'enzyme';

import { Switcher } from 'components/_basic/Switcher';
import { Default } from 'components/_basic/Switcher/Switcher.stories';

describe('<Switcher />', (): void => {
  test('should make a snapshot of component', (): void => {
    const wrapper = shallow(<Switcher />);
    const component = wrapper.getElement();

    expect(component).toMatchSnapshot();
  });

  test('should render Switcher from storybook', (): void => {
    const wrapper = shallow(<Default />);

    expect(wrapper.find(Switcher).length).toBe(1);
  });
});
