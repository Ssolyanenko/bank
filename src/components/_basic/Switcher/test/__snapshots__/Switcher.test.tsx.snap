// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switcher /> should make a snapshot of component 1`] = `
<div
  className="switcher"
>
  <Styled(ForwardRef(Switch))
    inputProps={
      {
        "aria-label": "primary checkbox",
      }
    }
  />
</div>
`;
